force-app/main/default/classes/StaticTask.cls

```diff
@@ -18,85 +16,90 @@ public with sharing class StaticTask {
       Contact parsedContact = new Contact();
-      List<String> contactSeparatedByComaString = getContactSeparatedByComaString(contactAsString);
+      List<String> contactSeparatedByComaString = getContactSeparatedByComaString(
+        contactAsString
+      );
       setContactToMap(contactSeparatedByComaString, parsedContact);
     }
   }
 
-  public static void setContactToMap(List<String> contactSeparatedByComaString, Contact parsedContact){
+  public static void setContactToMap(
+    List<String> contactSeparatedByComaString,
+    Contact parsedContact
+  ) {
     for (String element : contactSeparatedByComaString) {
       setFirstNameField(element, parsedContact);
       setLastNameField(element, parsedContact);
       setEmailField(element, parsedContact);
 
-      if(isEmailNotEmpty(element)){
+      if (isEmailNotEmpty(element)) {
         addContactToMap(parsedContact, emailToContactMap);
       }
     }
   }
 
-  public static void setEmailField(String elem, Contact contact){
+  public static void setEmailField(String elem, Contact contact) {
     if (isEmail(elem)) {
-      contact.Email = getFieldValueFromStr(elem); 
+      contact.Email = getFieldValueFromStr(elem);
     }
   }
 
-  public static void setFirstNameField(String elem, Contact contact){
+  public static void setFirstNameField(String elem, Contact contact) {
     if (isFirstName(elem)) {
       contact.FirstName = getFieldValueFromStr(elem);
     }
   }
 
-  public static void setLastNameField(String elem, Contact contact){
+  public static void setLastNameField(String elem, Contact contact) {
     if (isLastName(elem)) {
       contact.LastName = getFieldValueFromStr(elem);
     }
   }
 
-  public static void addContactToMap(Contact contact, Map<String, Contact> mapCollection) {
-    mapCollection.put(
-      contact.Email,
-      contact
-    );
+  public static void addContactToMap(
+    Contact contact,
+    Map<String, Contact> mapCollection
+  ) {
+    mapCollection.put(contact.Email, contact);
   }
 
-  public static Boolean isEmailNotEmpty(String elem){
+  public static Boolean isEmailNotEmpty(String elem) {
     if (isEmail(elem) && String.isNotBlank(elem.split(':')[1])) {
       return true;
     }
 
     return false;
   }
 
-  public static List<String> getContactSeparatedByComaString(String str){
+  public static List<String> getContactSeparatedByComaString(String str) {
     return str.split(',');
   }
 
-  public static List<String> getContactsAsStringList(String str){
+  public static List<String> getContactsAsStringList(String str) {
     return str.split(';');
   }
 
-  public static String getFieldValueFromStr(String str){
+  public static String getFieldValueFromStr(String str) {
     String result = str.split(':')[1];
     return trimm(result);
   }
 
-  public static Boolean isFirstName(String element){
+  public static Boolean isFirstName(String element) {
     return element.contains('FirstName');
```

Comment:
This method and the next two are almost the same. Can it be refactored for less code duplication?

ALso, is it possible to refactor this in such way, that we can easily add new fields? For example, we want to add Age, but we don't want to change the whole thing drastically.

_Hint_: think of collections